src/lib.rs

```diff
@@ -5,34 +5,33 @@ use bevy::{
     prelude::*,
     reflect::{FromType, GetTypeRegistration, TypeRegistry, TypeRegistryInternal},
 };
-use ggrs::{Config, PlayerHandle};
+use ggrs::{Config, InputStatus, P2PSession, PlayerHandle, SpectatorSession, SyncTestSession};
 use ggrs_stage::GGRSStage;
 use parking_lot::RwLock;
-use reflect_resource::ReflectResource;
+// use reflect_resource::ReflectResource;
 use std::sync::Arc;
 
 pub use ggrs;
 
 pub(crate) mod ggrs_stage;
-pub(crate) mod reflect_resource;
+// pub(crate) mod reflect_resource;
 pub(crate) mod world_snapshot;
 
 /// Stage label for the Custom GGRS Stage.
 pub const GGRS_UPDATE: &str = "ggrs_update";
 const DEFAULT_FPS: usize = 60;
 
 /// Defines the Session that the GGRS Plugin should expect as a resource.
-pub enum SessionType {
-    SyncTestSession,
-    P2PSession,
-    SpectatorSession,
+#[derive(Resource)]
+pub enum Session<T: Config> {
+    SyncTestSession(SyncTestSession<T>),
+    P2PSession(P2PSession<T>),
+    SpectatorSession(SpectatorSession<T>),
 }
 
-impl Default for SessionType {
-    fn default() -> Self {
-        SessionType::SyncTestSession
-    }
-}
+// TODO: more specific name to avoid conflicts?
```

Comment:
I think we should probably think about what to name this... It's public API that will be used, and `Inputs` quite easily conflicts with something else... and is very similar to Bevy's `Input`.

Reply:
I think I agree, but at t he same time it's namespaced so it's not a very big issue. What do you think of `SessionInputs` OR `SessionFrameInput` ? Also this could use some type documentation.

Reply:
yeah, it's easy enough to fix... More thinking about it being easy to read.

We have prefixed `GGRSError`, `GGRSConfig`. And some have `P2P` as prefix (`P2PSession`).

1. `GGRSInputs`
2. `P2PInputs`
3. `PlayerInputs` (matches `PlayerHandle` which is used to index this resource)
4. `SessionInputs`

Just spitballing. Don't really have a strong opinion. I'm also ok with `Inputs`. What do you think @gschup ?

definitely agree it could use some documentation!

Reply:
properly naming the Inputs is tough, but I think we should definitely not name it `Inputs`. I like `PlayerInputs` quite a bit. I would shy away from naming it `GGRSInputs` as this struct does not come from the GGRS crate itself.